data-prepper-core/integrationTest.gradle

```diff
@@ -169,7 +189,44 @@ task rawSpanEndToEndTest(type: Test) {
     classpath = sourceSets.integrationTest.runtimeClasspath
 
     filter {
-        includeTestsMatching "com.amazon.dataprepper.integration.EndToEndRawSpanTest*"
+        includeTestsMatching "com.amazon.dataprepper.integration.EndToEndRawSpanTest.testPipelineEndToEnd*"
+    }
+
+    finalizedBy stopOdfeDockerContainer
+    def stopDataPrepper1Task = stopDataPrepperDockerContainer(startDataPrepper1Task as DockerStartContainer)
+    def stopDataPrepper2Task = stopDataPrepperDockerContainer(startDataPrepper2Task as DockerStartContainer)
+    finalizedBy stopDataPrepper1Task
+    finalizedBy stopDataPrepper2Task
+    finalizedBy removeDataPrepperDockerContainer(stopDataPrepper1Task as DockerStopContainer)
+    finalizedBy removeDataPrepperDockerContainer(stopDataPrepper2Task as DockerStopContainer)
+    finalizedBy removeDataPrepperNetwork
+}
+
+task rawSpanCompatibilityEndToEndTest(type: Test) {
+    dependsOn build
+    dependsOn startOdfeDockerContainer
+    def createDataPrepper1Task = createDataPrepperDockerContainer(
+            "rawSpanDataPrepperFromBuild", "data-prepper1", 21890, 4900, "/app/${RAW_SPAN_PIPELINE_YAML}")
+    def createDataPrepper2Task = createDataPrepperDockerContainerFromPullImage(
+            "rawSpanDataPrepperFromPull", "data-prepper2", 21891, 4901, "src/integrationTest/resources/${RAW_SPAN_PIPELINE_NO_TRACE_GROUP_YAML}")
```

Comment:
Shouldn't these be using the same configuration yaml?

Also would we expect the test to pass given that Prepper2 is missing both a TraceGroupPrepper and PeerForwarder? Or are all requests being routed to Prepper1?

Reply:
As you noticed TraceGroupPrepper is missing and thus the separation of config yaml. The current testing logic is to send a portion of traceGroup with root span through the pulled data-prepper and then the data-prepper in the main branch should be able to pull root span from ES backend by the TraceGroupPrepper.

Reply:
Ah, I forgot that the latest released image does not have that prepper. So `RAW_SPAN_PIPELINE_NO_TRACE_GROUP_YAML` exists to be compatible with the latest docker image, while `RAW_SPAN_PIPELINE_YAML` is compatible with the local build.

Could we instead have _both_ running containers use a config file which is compatible with the most recent release (i.e. both containers using PeerForwarder instead of one having a TraceGroupPrepper)? The intent would be to simulate a user upgrading their deployment without updating their config. 

The test should still pass if PeerForwarders are introduced. `RAW_SPAN_PIPELINE_NO_TRACE_GROUP_YAML` would be something more like `RAW_SPAN_PIPELINE_LATEST_RELEASE_COMPATIBLE` (name tbd).

Reply:
Sure. Sounds a good strategy.